### PR TITLE
feat: upgrade vault-operator

### PR DESCRIPTION
**Pull request by j-zimnowoda**
#291

New vault operator contains functionality to automatically fill JWT issuer at Vault.
This is required since k8s 1.21 
Read more: https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#service-account-issuer-discovery

fixes #331

### CODE REVIEW

File: values/vault-operator/vault-operator.gotmpl [hunk 1]
```diff
@@ -5,3 +5,14 @@ global:
   imagePullSecrets:
     - otomi-pullsecret-global
 {{- end }}
+
+podSecurityContext:
+  runAsNonRoot: true
```
Comment (Morriz) [inline values/vault-operator/vault-operator.gotmpl:10]: `runAsNonRoot` is redundant when `runAsUser` is specified other than `0`

File: values/vault-operator/vault-operator.gotmpl [hunk 2]
```diff
@@ -5,3 +5,14 @@ global:
...
+  runAsUser: 65534
+  runAsGroup: 65534
+  fsGroup: 1337
+
+securityContext:
```
Comment (Morriz) [inline values/vault-operator/vault-operator.gotmpl:15]: if this targets the same pod (it seems it does), then `securityContext` is not needed when `podSecurityContext` is provided

**Review by Morriz — CHANGES_REQUESTED** [review]

File: values/vault-operator/vault-operator.gotmpl (hunk 2 above)
Comment (j-zimnowoda) [reply values/vault-operator/vault-operator.gotmpl]: done

File: values/vault-operator/vault-operator.gotmpl (hunk 1 above)
Comment (j-zimnowoda) [reply values/vault-operator/vault-operator.gotmpl]: done

**Review by Morriz — APPROVED** [review]

File: charts/vault-operator/README.md [hunk 3]
```diff
@@ -34,10 +34,13 @@ helm upgrade --install vault-operator banzaicloud-stable/vault-operator
 
 To install the chart backed with a cluster-wide Etcd Operator, use the following:
 
+**Please note that the etcd-operator has been deprecated for a long time (chart and code as well), and the last version of the chart is not installable on Kubernetes 1.22 and onwards. We don't offer any kind of support for etcd-operator.**
```
Comment (svatwork) [inline charts/vault-operator/README.md:37]: Actually this is a more recently maintained project if you need it: https://github.com/cilium/cilium-etcd-operator.

File: charts/vault-operator/templates/_helpers.tpl [hunk 4]
```diff
@@ -34,9 +34,12 @@ Create chart name and version as used by the chart label.
 {{/*
 Overrideable version for container image tags.
 */}}
-{{- define "vault-operator.bank-vaults.version" -}}
+{{- define "vault-operator.vault-operator.version" -}}
 {{- .Values.image.tag | default (printf "%s" .Chart.AppVersion) -}}
```
Comment (svatwork) [inline charts/vault-operator/templates/_helpers.tpl:38]: I assume semver of Helm charts (`.Chart.Appversion`) is not equal to `.Values.image.tag`, right? Is this a good default?

**Review by svatwork — APPROVED** [review]

File: charts/vault-operator/README.md (hunk 3 above)
Comment (j-zimnowoda) [reply charts/vault-operator/README.md:37]: Hi, I removed etcd from chart dependencies and we are not using it for our setup.
I also reported a bug regarding etcd.

File: charts/vault-operator/templates/_helpers.tpl (hunk 4 above)
Comment (j-zimnowoda) [reply charts/vault-operator/templates/_helpers.tpl:38]: The default falls back to `.Chart.AppVersion`. Seems to be right

**Pull request merged**